src/json-patch-duplex.ts

```diff
@@ -324,61 +322,43 @@ module jsonpatch {
     if (callback) {
       observer.callback = callback;
       observer.next = null;
-      var intervals = this.intervals || [100, 1000, 10000, 60000];
-      if (intervals.push === void 0) {
-        throw new OriginalError("jsonpatch.intervals must be an array");
-      }
-      var currentInterval = 0;
 
-      var dirtyCheck = function () {
-        generate(observer);
-      };
-      var fastCheck = function () {
-        clearTimeout(observer.next);
-        observer.next = setTimeout(function () {
-          dirtyCheck();
-          currentInterval = 0;
-          observer.next = setTimeout(slowCheck, intervals[currentInterval++]);
-        }, 0);
+      var dirtyCheck = () => {
+          generate(observer);
       };
-      var slowCheck = function () {
-        dirtyCheck();
-        if (currentInterval == intervals.length)
-          currentInterval = intervals.length - 1;
-        observer.next = setTimeout(slowCheck, intervals[currentInterval++]);
+      var fastCheck = () => {
+          clearTimeout(observer.next);
+          observer.next = setTimeout(() => {
+              dirtyCheck();
+          });
```

Comment:
@aloscha Have you tested it with some Polymer binding example (for example PuppetJs demo) that we do not need another single check after some delay?